### PR TITLE
Generalized cross-message types towards arbitrary cross-msgs

### PR DESCRIPTION
**Pull request by adlrocha**
- Switched `Fund` and `Release` cross-message types to `TopDown` and `BottomUp` respectively.
- Generalized the application of cross messages.
- Use of f04 addresses in messages to determine message type.

### CODE REVIEW

**Comment by adlrocha** [comment]
@dnkolegov, may I have a super quick review?

File: chain/consensus/hierarchical/actors/sca/sca_actor.go [hunk 1]
```diff
@@ -451,8 +453,48 @@ func SecpBLSAddr(rt runtime.Runtime, raw address.Address) address.Address {
...
+		rt.StateTransaction(&st, func() {
+			// Create releaseMsg and include in currentwindow checkpoint
+			st.releaseMsg(rt, value, secpAddr)
+		})
+		return nil
+	*/
+}
+
+// MsgParams determines the cross message to apply.
+type MsgParams struct {
```
[the remark this answers is not in the document]
Comment (dnkolegov) [reply chain/consensus/hierarchical/actors/sca/sca_actor.go:497]: Maybe `CrossMsg`?

File: chain/consensus/hierarchical/types.go [hunk 2]
```diff
@@ -78,3 +68,38 @@ var _ abi.Keyer = SubnetKey("")
 func (id SubnetKey) Key() string {
 	return string(id)
 }
+
+func CommonParent(from, to address.SubnetID) (address.SubnetID, int) {
+	s1 := strings.Split(from.String(), "/")
+	s2 := strings.Split(to.String(), "/")
+	if len(s1) < len(s2) {
+		a := s1
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/types.go:76]: why not s1, s2 = s2, s1 ?

File: chain/consensus/hierarchical/actors/sca/sca_actor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (adlrocha) [reply chain/consensus/hierarchical/actors/sca/sca_actor.go:503]: I totally get your point. Unfortunately CrossMsg is used in other contexts through the code, would `CrossMsgParam` work?

File: chain/consensus/hierarchical/types.go (hunk 2 above)
Comment (adlrocha) [reply chain/consensus/hierarchical/types.go:76]: Good catch! I forgot to change it after doing some tests. Thanks!

File: chain/consensus/hierarchical/actors/sca/sca_actor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnkolegov) [reply chain/consensus/hierarchical/actors/sca/sca_actor.go:503]: Yes, sounds better.

**Pull request merged**